src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6567,13 +6567,19 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 ASR::ttype_t* arg_type = nullptr;
                 if( ASR::is_a<ASR::Var_t>(*args[0].m_value) ) {
                     ASR::Var_t* arg_Var = ASR::down_cast<ASR::Var_t>(args[0].m_value);
-                    if( ASR::is_a<ASR::Variable_t>(*arg_Var->m_v) ) {
-                        arg_type = ASR::down_cast<ASR::Variable_t>(arg_Var->m_v)->m_type;
-                    } else if( ASR::is_a<ASR::StructType_t>(*arg_Var->m_v) ) {
+                    ASR::symbol_t* arg_Var_m_v = ASRUtils::symbol_get_past_external(arg_Var->m_v);
+                    if( ASR::is_a<ASR::Variable_t>(*arg_Var_m_v) ) {
+                        // TODO: Import the underlying struct if arg_type is of Struct type
+                        // Ideally if a variable of struct type is being imported then its underlying type
+                        // should also be imported automatically. However, the naming of the
+                        // underlying struct type might lead to collisions, so importing the type
+                        // here seems like a better choice. Should be done later when the case arises.
+                        arg_type = ASR::down_cast<ASR::Variable_t>(arg_Var_m_v)->m_type;
```

Comment:
This is related to https://docs.lfortran.org/en/asr/asr_nodes/expression_nodes/Var/. We already allow Var to point to a Function (symbol node), so that we can assign the function to a function pointer. This is almost identical: allow the structure itself to be passed as an argument to sizeof(). So Var could also point to Struct (=what we currently call StructType). We'll add StructType as a type, just like FunctionType.